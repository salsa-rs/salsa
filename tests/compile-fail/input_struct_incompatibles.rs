#[salsa::input(returns(ref))]
struct InputWithRetRef(u32);

#[salsa::input(specify)]
struct InputWithSpecify(u32);

#[salsa::input(no_eq)]
struct InputNoWithEq(u32);

#[salsa::input(db = Db)]
struct InputWithDb(u32);

#[salsa::input(recover_fn = recover)]
struct InputWithRecover(u32);

#[salsa::input(lru =12)]
struct InputWithLru(u32);

#[salsa::input(revisions = 12)]
struct InputWithRevisions(u32);

#[salsa::input]
struct InputWithTrackedField {
    #[tracked]
    field: u32,
}

fn main() {}
