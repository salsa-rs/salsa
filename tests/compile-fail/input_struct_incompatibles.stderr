error: `returns` option not allowed here
 --> tests/compile-fail/input_struct_incompatibles.rs:1:16
  |
1 | #[salsa::input(returns(ref))]
  |                ^^^^^^^

error: `specify` option not allowed here
 --> tests/compile-fail/input_struct_incompatibles.rs:4:16
  |
4 | #[salsa::input(specify)]
  |                ^^^^^^^

error: `no_eq` option not allowed here
 --> tests/compile-fail/input_struct_incompatibles.rs:7:16
  |
7 | #[salsa::input(no_eq)]
  |                ^^^^^

error: `db` option not allowed here
  --> tests/compile-fail/input_struct_incompatibles.rs:10:16
   |
10 | #[salsa::input(db = Db)]
   |                ^^

error: unrecognized option `recover_fn`
  --> tests/compile-fail/input_struct_incompatibles.rs:13:16
   |
13 | #[salsa::input(recover_fn = recover)]
   |                ^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/input_struct_incompatibles.rs:16:16
   |
16 | #[salsa::input(lru =12)]
   |                ^^^

error: `revisions` option not allowed here
  --> tests/compile-fail/input_struct_incompatibles.rs:19:16
   |
19 | #[salsa::input(revisions = 12)]
   |                ^^^^^^^^^

error: `#[tracked]` cannot be used with `#[salsa::input]`
  --> tests/compile-fail/input_struct_incompatibles.rs:24:5
   |
24 | /     #[tracked]
25 | |     field: u32,
   | |______________^

error: cannot find attribute `tracked` in this scope
  --> tests/compile-fail/input_struct_incompatibles.rs:24:7
   |
24 |     #[tracked]
   |       ^^^^^^^
   |
help: consider importing one of these attribute macros
   |
1  + use salsa::tracked;
   |
1  + use salsa_macros::tracked;
   |
